tests/test_scoped_nodes.py

```diff
@@ -1295,7 +1295,7 @@ def func(arg1, arg2):
             self.assertIsInstance(inferred[0], BoundMethod)
         inferred = list(Instance(cls).igetattr("m4"))
         self.assertEqual(len(inferred), 1)
-        self.assertIsInstance(inferred[0], nodes.FunctionDef)
+        self.assertIsInstance(inferred[0], BoundMethod)
```

Comment:
I think this change is incorrect, and therefore shows something is up with this PR.

To reproduce:
```python
class Clazz(object): ...


def func(arg1, arg2):
    "function that will be used as a method"
    return arg1.value + arg2


Clazz.m3 = func
inst = Clazz()
inst.m4 = func

print(Clazz.m3)
print(inst.m3)
print(inst.m4)
```


```shell
<function func at 0x109582340>
<bound method func of <__main__.Clazz object at 0x1095ab9b0>>
<function func at 0x109582340>
```

This should remain a `FunctionDef` and not a `BoundMethod`.